iroha-cpp/services/Query.cpp

```diff
@@ -0,0 +1,163 @@
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_account_assets();
+    pb_query_mut->set_account_id(account_id);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getAccountDetail(const std::string& account_id)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_account_detail();
+    pb_query_mut->set_account_id(account_id);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getAccountTransactions(const std::string& account_id)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_account_transactions();
+    pb_query_mut->set_account_id(account_id);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getAccountAssetTransactions(const std::string& account_id, const std::string& asset_id)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_account_asset_transactions();
+    pb_query_mut->set_account_id(account_id);
+    pb_query_mut->set_asset_id(asset_id);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getTransactions(const std::vector<std::string>& tx_hashes)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_transactions();
+    std::for_each(tx_hashes.begin(),
+                  tx_hashes.end(),
+                  [&pb_query_mut](auto tx_hash)
+    {
+        auto adder = pb_query_mut->add_tx_hashes();
+        //        *adder = tx_hash.to_hexstring();    // TODO string do hexstring()
+    });
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getSignatories(const std::string& account_id)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_signatories();
+    pb_query_mut->set_account_id(account_id);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getAssetInfo(const std::string& asset_id)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_asset_info();
+    pb_query_mut->set_asset_id(asset_id);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getRoles()
+{
+    iroha::protocol::Query pb_query;
+    pb_query.mutable_payload()->mutable_get_roles();
+    populateQueryMetaData(pb_query);
+
+    queryProto = pb_query;
+    return *this;
+}
+
+Query& Query::getRolePermissions(const std::string& role_id)
+{
+    iroha::protocol::Query pb_query;
+    populateQueryMetaData(pb_query);
+
+    auto pb_query_mut = pb_query.mutable_payload()->mutable_get_role_permissions();
+    pb_query_mut->set_role_id(role_id);
+
+    queryProto = pb_query;
```

Comment:
I see that:
```
iroha::protocol::Query queryProto;
```
so do we need extra variable: `pb_query`?

Reply:
Outdated comment.